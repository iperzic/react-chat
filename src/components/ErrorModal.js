import React from 'react';
import PropTypes from 'prop-types';

import './ErrorModal.css';

const ErrorModal = ({ message }) => (
  <div className="Modal">
    <div className="modal-content">
      {message}
    </div>
  </div>
);

ErrorModal.propTypes = {
  message: PropTypes.string.isRequired,
};

export default ErrorModal;
