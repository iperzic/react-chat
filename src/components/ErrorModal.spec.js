import React from 'react';
import { shallow } from 'enzyme';

import ErrorModal from './ErrorModal';

describe('ErrorModal', () => {
  let component;
  let message;

  beforeEach(() => {
    message = 'An error has occurred';
    component = shallow(<ErrorModal
      message={message}
    />);
  });

  it('should display the error modal with provided text', () => {
    expect(component.getElement()).toMatchSnapshot();
  });
});
