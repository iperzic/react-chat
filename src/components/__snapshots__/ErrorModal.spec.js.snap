// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal should display the error modal with provided text 1`] = `
<div
  className="Modal"
>
  <div
    className="modal-content"
  >
    An error has occurred
  </div>
</div>
`;
